datafusion/core/src/physical_plan/file_format/row_filter.rs

```diff
@@ -228,11 +230,32 @@ fn size_of_columns(columns: &[usize], metadata: &ParquetMetaData) -> Result<usiz
 /// For a given set of `Column`s required for predicate `Expr` determine whether all
 /// columns are sorted. Sorted columns may be queried more efficiently in the presence of
 /// a PageIndex.
-fn columns_sorted(_columns: &[usize], _metadata: &ParquetMetaData) -> Result<bool> {
-    // TODO How do we know this?
+fn columns_sorted(columns: &[usize], metadata: &ParquetMetaData) -> Result<bool> {
```

Comment:
I may be misreading the spec, but the documentation on 

https://github.com/apache/parquet-format/blob/54e53e5d7794d383529dd30746378f19a12afd58/src/main/thrift/parquet.thrift#L948-L959


```

  /**
   * Stores whether both min_values and max_values are orderd and if so, in
   * which direction. This allows readers to perform binary searches in both
   * lists. Readers cannot assume that max_values[i] <= min_values[i+1], even
   * if the lists are ordered.
   */
  4: required BoundaryOrder boundary_order
```

Implies to me that `BoundaryOrder` is related to the ordering of the elements in `min_values` and `max_values` lists of the metadata and not the ordering of the actual data in the pages. Give the comments on `min_values` and `max_values` it seems like those can have elements that are not in the actual file itself


So I wonder can one conclude that the data is sorted even if `BoundaryOrder` is specified? 🤔

Reply:
>So I wonder can one conclude that the data is sorted even if BoundaryOrder is specified? 🤔

Mostly we can assume is sorted, like in `parquet-testing`  use `parquet-tools column-index ./alltypes_tiny_pages.parquet` we can find all col with `BoundaryOrder ` are sorted.😂

As we assume `columns_sorted ` as
```
Sorted columns may be queried more efficiently in the presence of a PageIndex.
```
I think use Page-index `BoundaryOrder ` is reasonable🤔

Reply:
Maybe i can try to find the java code, and it shows how to generate the `boundary_order` for one ColumnIndex.